pkg/controller/controller.go

```diff
@@ -110,26 +110,45 @@ type controller struct {
 	brokerLister           listers.BrokerLister
 	serviceClassLister     listers.ServiceClassLister
 	instanceLister         listers.InstanceLister
+	bindingLister          listers.BindingLister
+	brokerQueue            workqueue.RateLimitingInterface
+	serviceClassQueue      workqueue.RateLimitingInterface
+	instanceQueue          workqueue.RateLimitingInterface
+	bindingQueue           workqueue.RateLimitingInterface
 }
 
 // Run runs the controller until the given stop channel can be read from.
-func (c *controller) Run(stopCh <-chan struct{}) {
+func (c *controller) Run(workers int, stopCh <-chan struct{}) {
 	defer runtimeutil.HandleCrash()
+	defer func() {
+		c.brokerQueue.ShutDown()
+		c.serviceClassQueue.ShutDown()
+		c.instanceQueue.ShutDown()
+		c.bindingQueue.ShutDown()
+	}()
```

Comment:
Instead of putting this in a deferred func, why not put it under the log line that reads:

```go
glog.Info("Shutting down service-catalog controller")
```

Doing so makes these lines explicitly reliant on the `stopCh`